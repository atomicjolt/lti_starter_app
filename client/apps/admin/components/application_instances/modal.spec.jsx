import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Modal from './modal';

const mockStore = configureStore([]);
const store = mockStore({
  settings: {
    canvas_callback_url: 'https://www.example.com',
  },
});

describe('application instance modal', () => {
  let result;
  let props;
  let closed;
  let saved;
  const name = 'the application';

  beforeEach(() => {
    saved = false;
    closed = false;
    props = {
      isOpen: true,
      closeModal: () => { closed = true; },
      sites: {},
      save: () => { saved = true; },
      applicationInstance: {
        id: 2,
        config: 'config string',
        site: {
          id: 3,
        },
      },
      application: {
        id: 1,
        name
      }
    };
    result = TestRenderer.create(
      <Provider store={store}>
        <Modal {...props} />
      </Provider>
    );
  });

  it('match the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the changing of the siteModalOpen state open', () => {
    expect(result.root.state.siteModalOpen).toBeFalsy();
    result.root.newSite();
    expect(result.root.state.siteModalOpen).toBeTruthy();
  });

  it('handles the changing of the siteModalOpen state to close', () => {
    expect(closed).toBeFalsy();
    expect(result.root.state.siteModalOpen).toBeFalsy();
    result.root.newSite();
    expect(result.root.state.siteModalOpen).toBeTruthy();
    result.root.closeModal();
    expect(closed).toBeTruthy();
    expect(result.root.state.siteModalOpen).toBeFalsy();
  });

  it('handles the save function', () => {
    expect(saved).toBeFalsy();
    expect(result.root.state.siteModalOpen).toBeFalsy();
    result.root.newSite();
    expect(result.root.state.siteModalOpen).toBeTruthy();
    result.root.save();
    expect(saved).toBeTruthy();
  });
});
