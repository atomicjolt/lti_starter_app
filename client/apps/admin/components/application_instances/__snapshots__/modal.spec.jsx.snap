// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application instance modal match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    application={
      Object {
        "id": 1,
        "name": "the application",
      }
    }
    applicationInstance={
      Object {
        "config": "config string",
        "id": 2,
        "site": Object {
          "id": 3,
        },
      }
    }
    closeModal={[Function]}
    isOpen={true}
    save={[Function]}
    sites={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ariaHideApp": true,
      "bodyOpenClassName": "ReactModal__Body--open",
      "children": Array [
        <h2
          className="c-modal__title"
        >
          Update
           
          the application
           Instance
        </h2>,
        <Form
          closeModal={[Function]}
          config="config string"
          id={2}
          isUpdate={true}
          newSite={[Function]}
          onChange={[Function]}
          save={[Function]}
          site={
            Object {
              "id": 3,
            }
          }
          site_id="3"
          sites={Object {}}
        />,
        <Connect(SiteModal)
          closeModal={[Function]}
          isOpen={false}
        />,
      ],
      "className": "c-modal c-modal--settings is-open",
      "closeTimeoutMS": 0,
      "contentLabel": "Application Instances Modal",
      "isOpen": true,
      "onRequestClose": [Function],
      "overlayClassName": "c-modal__background",
      "parentSelector": [Function],
      "portalClassName": "ReactModalPortal",
      "shouldCloseOnEsc": true,
      "shouldCloseOnOverlayClick": true,
      "shouldFocusAfterRender": true,
      "shouldReturnFocusAfterClose": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Update",
            " ",
            "the application",
            " Instance",
          ],
          "className": "c-modal__title",
        },
        "ref": null,
        "rendered": Array [
          "Update",
          " ",
          "the application",
          " Instance",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeModal": [Function],
          "config": "config string",
          "configParseError": undefined,
          "id": 2,
          "isUpdate": true,
          "newSite": [Function],
          "onChange": [Function],
          "save": [Function],
          "site": Object {
            "id": 3,
          },
          "site_id": "3",
          "sites": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "closeModal": [Function],
          "isOpen": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ariaHideApp": true,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": Array [
          <h2
            className="c-modal__title"
          >
            Update
             
            the application
             Instance
          </h2>,
          <Form
            closeModal={[Function]}
            config="config string"
            id={2}
            isUpdate={true}
            newSite={[Function]}
            onChange={[Function]}
            save={[Function]}
            site={
              Object {
                "id": 3,
              }
            }
            site_id="3"
            sites={Object {}}
          />,
          <Connect(SiteModal)
            closeModal={[Function]}
            isOpen={false}
          />,
        ],
        "className": "c-modal c-modal--settings is-open",
        "closeTimeoutMS": 0,
        "contentLabel": "Application Instances Modal",
        "isOpen": true,
        "onRequestClose": [Function],
        "overlayClassName": "c-modal__background",
        "parentSelector": [Function],
        "portalClassName": "ReactModalPortal",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Update",
              " ",
              "the application",
              " Instance",
            ],
            "className": "c-modal__title",
          },
          "ref": null,
          "rendered": Array [
            "Update",
            " ",
            "the application",
            " Instance",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeModal": [Function],
            "config": "config string",
            "configParseError": undefined,
            "id": 2,
            "isUpdate": true,
            "newSite": [Function],
            "onChange": [Function],
            "save": [Function],
            "site": Object {
              "id": 3,
            },
            "site_id": "3",
            "sites": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeModal": [Function],
            "isOpen": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
